lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -90,7 +98,15 @@ class << self
           #         should be stored in a file list file.
           #
           def input_output_paths_use_filelist?(object)
-            object.project.object_version.to_i >= 50
+            unless object.project.root_object.compatibility_version.nil?
+              version_match = object.project.root_object.compatibility_version.match(/Xcode ([0-9]*.[0-9]*)/)
+              compatibility_version = version_match[1].to_f unless version_match.nil? || version_match.length < 2
+            end
+            if compatibility_version.to_f > 0
+              compatibility_version.to_f >= MIN_FILE_LIST_COMPATIBILITY_VERSION
```

Comment:
I'd like to avoid floating point comparisons here - can we compare `Pod::Version.new(compatibility_version) >= Pod::Version.new(MIN_FILE_LIST_COMPATIBILITY_VERSION)` instead?